pkg/osutil/osutil.go

```diff
@@ -245,6 +245,10 @@ func MkdirAll(dir string) error {
 	return os.MkdirAll(dir, DefaultDirPerm)
 }
 
+func Mkdir(path string) error {
```

Comment:
I would say just use the existing MkdirAll, that's what we used everywhere so far. It's simpler and more reliable. Most dirs are created once on process startup, so optimizing for "oh, we already created another sibling dir before, so for this one we could use Mkdir instead of MkdirAll" is not worth it. It can also lead to bugs later if code is shuffled a bit, or dir structure is changed. And the bug may go unnoticed for some time if you happen to have dirs already created locally (which you will most likely have).